homeassistant/components/kef/media_player.py

```diff
@@ -304,3 +391,69 @@ def icon(self):
     async def async_media_next_track(self):
         """Send next track command."""
         await self._speaker.next_track()
+
+    async def update_dsp(self):
+        """Update the DSP settings."""
+        mode = await self._speaker.get_mode()
+        self._dsp = dict(
+            **mode._asdict(),
+            desk_db=await self._speaker.get_desk_db(),
+            wall_db=await self._speaker.get_wall_db(),
+            treble_db=await self._speaker.get_treble_db(),
+            high_hz=await self._speaker.get_high_hz(),
+            low_hz=await self._speaker.get_low_hz(),
+            sub_db=await self._speaker.get_sub_db(),
+        )
+        return self._dsp
+
+    @property
+    def device_state_attributes(self):
+        """Return the DSP settings of the KEF device."""
+        if self._dsp is None:
+            return dict(
+                dsp_settings="Run the `kef.update_dsp` service to get the DSP settings as attributes."
+            )
```

Comment:
Please use `{…}` to define dicts. It's faster.

Also, it seems like this is a bit weird. Let's not make this ever happen and instead update dsp inside `async_added_to_hass`

Reply:
How do I make sure that the `update_dsp` is only called whenever the speaker turns on for the first time?

Because only then it's available.

Reply:
Inside `async_update` call it automatically if `self._dsp` is None. Until then, just don't include it ?

Reply:
Right, of course! 😅